src/message.lisp

```diff
@@ -49,55 +92,185 @@
        :reader get-message-id)
    (text :initarg :text
          :reader get-text)
+   ;; Caption for file messages
+   (caption :initarg :caption
+            :reader get-caption)
    (chat :initarg :chat
          :reader get-chat)
    (entities :initarg :entities
              :initform nil
              :reader get-entities)
    (raw-data :initarg :raw-data
-             :reader get-raw-data)))
-
-(defclass reply (message)
-  ((reply-to-message :initarg :reply-to-message
-                     :reader get-reply-to-message)))
-
-(defclass forwarded (message)
-  ((forward-from :initarg :forward-from
+             :reader get-raw-data)
+   (forward-from :initarg :forward-from
                  :reader get-forward-from)
    (forward-sender-name :initarg :forward-sender-name
                         :reader get-forward-sender-name)
    (forward-from-chat :initarg :forward-from-chat
                       :reader get-forward-from-chat)))
 
+(defmethod initialize-instance :after ((message message) &key data &allow-other-keys)
+  (when data
+    (setf (slot-value message 'id) (getf data :|message_id|)
+          (slot-value message 'text) (getf data :|text|)
+          (slot-value message 'chat) (make-chat (getf data :|chat|))
+          (slot-value message 'entities) (mapcar (lambda (item)
+                                                   (make-entity message item))
+                                                 (getf data :|entities|))
+          (slot-value message 'raw-data) data
+          (slot-value message 'forward-from-chat) (when (getf data :|forward_from_chat|)
+                                                    (make-chat (getf data :|forward_from_chat|)))
+          (slot-value message 'forward-from) (when (getf data :|forward_from_chat|)
```

Comment:
Here you are checking `forward_from_chat` item in the `data` as two lines before. Seems as a bug.

By the way, it is good idea to add `type` to slot declarations. For optional slots like `forward_from_chat` it should be `(or null chat)`.

Reply:
Yes, a typo it is!

The only reason I didn't add `type`s is not seeing them in the initial class definitions xD Will do!